File: spec/starting_position_spec.rb
require_relative "../lib/starting_position"

describe StartingPosition do
  
  position = StartingPosition.new

  it "has properly placed generals" do
    position.find(4, 0).class.should eql(General)
    position.find(4, 10).class.should eql(General)
  end
end

File: spec/starting_position_spec.js
var StartingPosition = require('../src/starting_position.js');

describe('StartingPosition', function() {
  var position;

  it('has pieces on squares', function() {
    position = new StartingPosition();
    var coordinates = Object.keys(position)[0];

    expect(position[coordinates].square).toBeDefined();
  });
});
